### ISSUE TITLE
Confusing alpha channel in ColorPicker

### ISSUE THREAD

**Issue by berruezo** [AvaloniaUI/Avalonia#10094]
**Describe the bug**
The alpha channel in the ColorPicker is a bit confusing: the value represented in the TextBox (where you can type or paste the HEX color) uses the ARGB form, but the order of the color sliders is RGBA.

Also, when the IsAlphaEnabled property is set to false, the alpha control is still visible (although disabled)

**Screenshots**
![image](https://user-images.githubusercontent.com/28261333/214866900-0229ca7e-044f-4bd5-9755-28976ca54377.png)

**Comment by robloo** [comment]
I did a deeper review here and don't think it can be "fixed" without causing other people to complain. Bottom line though, this issue shouldn't be a bug and is more of a discussion at this point.

** Hex formats ARGB vs RGBA **
 
 * [ARGB] This is a big one, XAML itself uses `#AARRGGBB` or `#ARGB` format. This started back in WPF and continues with Avalonia for compatibility. The hex color parser is obviously setup to handle only this format and can't be changed to auto-detect ARGB or RGBA. For this reason alone we likely cannot change the hex input box default -- it should match other cases in Avalonia.
 * [ARGB] Both the WPF `Color` struct and Avalonia's `Color` struct use ARGB format in the constructors and in `FromArgb()` methods. Again, this establishes the XAML convention.
 * [ARGB] WinUI ColorPicker : The WinUI ColorPicker was the original basis for this control in Avalonia. It is the reason for the "confusion" points you mentioned above. The WinUI ColorPicker has the same issues (however they refer to Alpha as Opacity and use a percentage)
    ![image](https://user-images.githubusercontent.com/17993847/215273802-7a581fa3-e6a4-4d5c-ac3e-8aa03f5f6b22.png)
 * [RGBA] CSS added (in-draft still...) the hex color notation with alpha in #RRGGBBAA format. I really, really, wish this aligned with XAML but it doesn't. XAML pre-dates CSS here by quite a bit and unfortunately they chose different formats. I would argue CSS format is actually better though as optional parameters should go at the end. Additionally, it matches with `rgba()` and `hsla()` notation parameter ordering.
    * https://drafts.csswg.org/css-color/#hex-notation
    * https://caniuse.com/css-rrggbbaa
 * [RGBA] In Avalonia 11 it is now possible to define colors using some CSS function notation like `rgba()` or `hsla()`. This was done for compatibility with CSS and what users are used to in that case. However, the alpha component is at the end and it does differ from hex values. This isn't so bad in this case because ordering is clear by the `rgba(` prefix itself.
 * [Other] Looking at photo editing apps like Gimp and PhotoShop, these apps don't even have an alpha channel in the color edit dialogs. That makes them useless as examples for this discussion.

This also relates to an older discussion: https://github.com/AvaloniaUI/Avalonia/pull/7724#discussion_r816374775.

I'm debating a few things here since there isn't a way to make everyone happy here. Someone is going to think it's wrong no matter what is done:

 1. Leave it alone, it matches XAML and the WinUI ColorPicker
 2. Add another property to actually control the color format RGBA or ARGB. That would allow XAML standard as default but also CSS RGBA format which is more popular now.
    * That opens up potential secondary issues though: Should it use existing Avalonia color format enums? (no, probably not) Should this control the alpha slider position as well?
 3. Rename the alpha slider to opacity like WinUI

**Alpha Slider Enabled/Visible**

I'm not planning to change the behavior here. `IsAlphaEnabled` and `IsAlphaVisible` allow controlling this just like Control.IsEnabled and Control.IsVisible. The concept is already established. If you want to hide the slider use `IsAlphaVsisible`.

That said, the hex value should be removing the alpha component when both IsAlphaEnabled=False and IsAlphaVisible=False. It currently isn't.

**Comment by timunie** [comment]
One addition: I think if you are happy if the sliders are matching the order of `#ARGB` this is easy to achieve if you provide your own control template for `ColorView`. Basically copy & paste the original and reorder the sliders. 

Original style to edit: https://github.com/AvaloniaUI/Avalonia/blob/30438713d1dbee8cb168520d3b2d25e79ba9e973/src/Avalonia.Controls.ColorPicker/Themes/Fluent/ColorView.xaml#L507-L731

**Comment by robloo** [comment]
I suppose another idea here is to change alpha hex format in Avalonia from XAML ARGB to CSS RGBA formats. That would fundamentally fix this once and for all (while making more sense too, optional parameter at the end).

However it would break a lot of apps in difficult to find ways. Don't think it would ever be considered a good idea for this reason.

**Comment by robloo** [comment]
I would like to add a new property to control the hex input format -- and only the hex input format (sliders will stay matching WinUI for now). That should be a good compromise and allow devs to use RGBA when they want to match CSS or the slider ordering.

However, naming the enum is tricky at it should be generalized more than just ARGB and RGBA (even though the hex input is only using those formats right now). I'm leaning towards something like the below. Feedback welcome.

```csharp
enum AlphaPosition
{
    // ARGB or AHSV, etc.
    Leading,

    // RGBA or HSVA, etc.
    Trailing
}

// In ColorView
public AlphaPosition HexAlphaPosition { get; set; }
```

**Comment by danipen** [comment]
> That is planned to be addressed; however, I still don't consider it a bug and its by design matching WinUI. Please comment on the original issue if you have any suggestions for the new enum.

@berruezo any thoughts on this?

**Comment by robloo** [comment]
I need some feedback here or I probably won't be implementing a change for a while. Its best not to create a new enum and property like this without some discussion/feedback. People usually have differences of opinion and right now it matches WinUI.

Additionally, if the author of the issue doesn't consider it important enough to engage I have to think it isnt enough of an issue to spend time on it for now. Everyone is busy. Please engage in the discussion if you want changes.

**Comment by danipen** [comment]
Pinging @berruezo for an alternative channel. He may answer tomorrow sorry for the delay.

**Comment by berruezo** [comment]
Sorry, I didn't get notified about the new replies in the thread

The enum looks good to me, it would allow us to have the same order for the sliders as for the hex value, which was the main point of concern. I find the name of the property and values pretty accurate with what it's doing.

**Comment by robloo** [comment]
I've added support for these changes in https://github.com/AvaloniaUI/Avalonia/pull/10439.

`ColorView` now has the `HexInputAlphaPosition` property which can be used to configure this to match XAML/WinUI (ARGB) or CSS (RGBA).

I decided to set the property default value to `AlphaComponentPosition.Trailing` which is what you initially wanted. The reasons are compelling:
 1. It matches the now more widely used CSS standard
 2. It matches the default slider ordering as shown to end-users

The only downside is it now differs by default than the WinUI control and XAML hex formats. However, it's better future proofed following CSS. It's also better to change it now before the control is released in a stable version of Avalonia. This can be switched back to the XAML/WinUI by setting the property directly on the control to `Leading`.
